misago/users/signals.py

```diff
@@ -1,6 +1,17 @@
+from datetime import timedelta
+
 from django.dispatch import Signal, receiver
+from django.contrib.auth import get_user_model
+from django.utils import timezone
+
+from django.db.models import Q
```

Comment:
This is much better than previous one, but still not right. It should be like this:

```python
from datetime import timedelta

from django.contrib.auth import get_user_model
from django.db.models import Q
from django.dispatch import Signal, receiver
from django.utils import timezone

from misago.conf import settings
from misago.core.utils import ANONYMOUS_IP
```